txcache/txByHashMap_test.go

```diff
@@ -14,56 +13,29 @@ func Test_newTxByHashMap(t *testing.T) {
 	require.NotNil(t, txByHashmap)
 }
 
-func createMockWrappedTx(txHash []byte) *WrappedTransaction {
-	return &WrappedTransaction{
-		Tx:     nil,
-		TxHash: txHash,
-	}
-}
-
-func createSliceMockWrappedTxs(txHashes [][]byte) []*WrappedTransaction {
-	wrappedTxs := make([]*WrappedTransaction, len(txHashes))
-	for i, txHash := range txHashes {
-		wrappedTxs[i] = createMockWrappedTx(txHash)
-	}
-
-	return wrappedTxs
-}
-
-func createMockTxHashes(numberOfTxs int) [][]byte {
-	txHashes := make([][]byte, numberOfTxs)
-	for i := 0; i < numberOfTxs; i++ {
-		txHashes[i] = []byte("txHash" + strconv.Itoa(i))
-	}
-
-	return txHashes
-}
-
-func addWrappedTxs(txByHash *txByHashMap, wrappedTxs []*WrappedTransaction, numberOfTxs int) {
+func addWrappedTxsConcurrently(txByHash *txByHashMap, wrappedTxs []*WrappedTransaction, numberOfTxs int) {
 	var wg sync.WaitGroup
 	wg.Add(numberOfTxs)
 
 	for _, wrappedTx := range wrappedTxs {
-		wrappedTxCopy := wrappedTx
-		go func() {
+		go func(wrappedTx *WrappedTransaction) {
 			defer wg.Done()
-			txByHash.addTx(wrappedTxCopy)
-		}()
+			txByHash.addTx(wrappedTx)
+		}(wrappedTx)
 	}
 
 	wg.Wait()
 }
 
-func removeWrappedTxs(txByHash *txByHashMap, wrappedTxs []*WrappedTransaction, numberOfTxs int) {
+func removeWrappedTxsConcurrently(txByHash *txByHashMap, wrappedTxs []*WrappedTransaction, numberOfTxs int) {
 	var wg sync.WaitGroup
 	wg.Add(numberOfTxs)
 
 	for _, wrappedTx := range wrappedTxs {
-		wrappedTxCopy := wrappedTx
-		go func() {
+		go func(wrappedTx *WrappedTransaction) {
 			defer wg.Done()
-			txByHash.removeTx(string(wrappedTxCopy.TxHash))
-		}()
+			txByHash.removeTx(string(wrappedTx.TxHash))
+		}(wrappedTx)
```

Comment:
:+1: 

Sorry for my previous comment - previous approach has good, as well (only the name was a bit ambiguous _in my opinion_).